packages/next-server/server/render.tsx

```diff
@@ -255,7 +259,8 @@ export async function renderToHTML(
   } = renderOpts
 
   await Loadable.preloadAll() // Make sure all dynamic imports are loaded
-  let isStaticPage = false
+  let isStaticPage = Boolean(pageConfig.experimentalPrerender)
```

Comment:
Should this be more strict?
```suggestion
  let isStaticPage = pageConfig.experimentalPrerender === true
```